app/services/force/soql/preference_service.rb

```diff
@@ -50,6 +60,15 @@ def app_preferences_for_listing_query(opts)
                .transform_results { |results| massage(results) }
       end
 
+      def app_preferences_for_listing_single_query(opts)
```

Comment:
we should add a comment to explain this too since it's pretty hacky. Let me know if you want me to write up the language from it.